### PR TITLE
Proc dep warnings

### PR DESCRIPTION
**Pull request by scopatz**
Some deprecation help for downstream code

### CODE REVIEW

**Comment by scopatz** [comment]
Ready for review @xonsh/xore

File: xonsh/proc.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""xonsh.proc is deprecated"""
+from xonsh.lazyasd import lazyobject
+from xonsh.tools import print_warning
+
+
+_WARNINGS_PRINTED = {}
+
+
+def _print_proc_warning(msg):
+    global _WARNINGS_PRINTED
```
Comment (daniel-shimon) [inline xonsh/proc.py:10]: Not needed (no assignment)

File: xonsh/proc.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    if msg not in _WARNINGS_PRINTED:
+        print_warning(msg)
+        _WARNINGS_PRINTED[msg] = True
```
Comment (daniel-shimon) [inline xonsh/proc.py:13]: Use a set instead

**Review by daniel-shimon — CHANGES_REQUESTED** [review]

File: xonsh/proc.py (hunk 1 above)
Comment (scopatz) [reply xonsh/proc.py:10]: That is technically correct, but I think this is good defensive coding / future-proofing.

**Comment by scopatz** [comment]
Thanks for the review @daniel-shimon! I think this is ready for review again @xonsh/xore (and is blocking a release)

**Review by gforsyth — APPROVED** [review]
looks good -- sorry for the delay @scopatz, I am buried in notifications

**Pull request merged**